substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/pecoff/cv/CVTypeRecord.java

```diff
@@ -69,17 +100,23 @@ int computeFullSize(int initialPos) {
     }
 
     int computeFullContents(byte[] buffer, int initialPos) {
+        assert sequenceNumber >= FIRST_TYPE_INDEX;
         int pos = initialPos + Short.BYTES; /* Save room for length short. */
         pos = CVUtil.putShort(type, buffer, pos);
         pos = computeContents(buffer, pos);
         /* Length does not include record length (2 bytes)) but does include end padding. */
         pos = alignPadded4(buffer, pos);
-        int length = (short) (pos - initialPos - Short.BYTES);
+        int length = pos - initialPos - Short.BYTES;
+        if (length > CV_TYPE_RECORD_MAX_SIZE) {
+            GraalError.shouldNotReachHere(String.format("Type record too large: %d (maximum %d) bytes: %s", length, CV_TYPE_RECORD_MAX_SIZE, this));
```

Comment:
We usually use `throw` in cases like this.
```suggestion
            throw GraalError.shouldNotReachHere(String.format("Type record too large: %d (maximum %d) bytes: %s", length, CV_TYPE_RECORD_MAX_SIZE, this));
```